indexer/xml_feature.cpp

```diff
@@ -0,0 +1,141 @@
+#include "indexer/xml_feature.hpp"
+
+#include "base/string_utils.hpp"
+#include "base/timer.hpp"
+
+#include "std/cstring.hpp"
+
+#include "3party/pugixml/src/pugixml.hpp"
+
+
+namespace
+{
+string ToString(m2::PointD const & p)
+{
+  ostringstream out;
+  out.precision(7);
+  out << p.x << ", " << p.y;
+  return out.str();
+}
+
+bool FromString(string const & str, m2::PointD & p)
+{
+  double x, y;
+  istringstream sstr(str);
+  sstr >> x;
+  sstr.get();
+  sstr.get();
+  sstr >> y;
+  p = {x, y};
+  return true;
+}
+
+void AddTag(string const & key, string const & value, pugi::xml_node & node)
+{
+  auto tag = node.append_child("tag");
+  tag.append_attribute("k") = key.data();
+  tag.append_attribute("v") = value.data();
+}
+
+pugi::xpath_node FindTag(pugi::xml_document const & document, string const & key)
+{
+  return document.select_node(("//tag[@k='" + key + "']").data());
+}
+} // namespace
+
+
+namespace indexer
+{
+XMLFeature::XMLFeature(): m_documentPtr(new pugi::xml_document) {}
+
+XMLFeature::XMLFeature(string const & xml):
```

Comment:
I suppose we'll create features from pugi::xml_node instead of string (there will be pugi xml loader which passes XML tree leaves - features - here).